pkg/generate/app/pipeline.go

```diff
@@ -91,7 +91,7 @@ func (p *Pipeline) NeedsDeployment(env Environment, name string) error {
 // Objects converts all the components in the pipeline into runtime objects
 func (p *Pipeline) Objects(accept, objectAccept Acceptor) (Objects, error) {
 	objects := Objects{}
-	if p.InputImage != nil && p.InputImage.AsImageStream && accept.Accept(p.InputImage) {
+	if p.InputImage != nil && accept.Accept(p.InputImage) {
```

Comment:
I think this is confusing what AsImageStream was intended for.  Originally, AsInputStream was "generate this as an image stream".  This is starting to use it as "reference a docker repository in a created image stream".  I don't think the two are related - AsInputStream should be set to true *by* the generation code code prior to getting in the pipeline.  There needs to be something on imageref that indicates that the resulting generated image stream comes from a docker repository or not (which is how you're using AsImageStream)